### ISSUE TITLE
Implement versioning of query input data format

### ISSUE THREAD

**Issue by daniel-thom** [dsgrid/dsgrid#204]

**Comment by elainethale** [comment]
Could this be the same as a dsgrid software version? Are we ready to make an initial release and start supporting backward compatibility (for the query input data format, but also all other config data formats)?

**Comment by daniel-thom** [comment]
I don't think so. We will likely change dsgrid versions far more often that the input data format.

**Comment by elainethale** [comment]
But should we just have a map of dsgrid version changes to input data format changes? I don't like the idea of each piece of input data having its own version number.

**Comment by daniel-thom** [comment]
I think it would be simpler to automatically encode a data format version into every query file. Any reader (person or machine) knows how to handle the data based on the version. We could even write scripts to upgrade the format if needed. The other way is OK with me, too.
